### PR TITLE
Connect discord for invite page

### PR DESCRIPTION
**Pull request by Devorein**
**Session Exist**
1. We get the user from the session. So no matter which wallet address you switch to it won't have any effect on the invite flow. We use the user from the session
2. After a successful discord connect, we log you as that user so clicking on accept Invite would add that user to the workspace

**Session Doesn't Exist**
1. Changing the wallet address will have an impact as before accepting the invite we create the user with that account (it won't create one if a user with that wallet address already exists and log you as that user)
2. If you connect to discord we fallback to **Session Exist**

### CODE REVIEW

File: components/invite/InviteLinkPage.tsx [hunk 1]
```diff
@@ -27,13 +28,21 @@ export default function InvitationPage ({ error, invite }: { error?: string, inv
...
+    }
+    if (invite) {
+      await charmClient.acceptInvite({ id: invite.id });
+      window.location.href = `/${invite.space.domain}`;
     }
-    await charmClient.acceptInvite({ id: invite!.id });
-    window.location.href = `/${invite!.space.domain}`;
   }
 
+  const spaceRole = user?.spaceRoles.find(_spaceRole => _spaceRole.spaceId === invite?.spaceId);
```
Comment (mattcasey) [inline components/invite/InviteLinkPage.tsx:40]: why is this needed?

File: components/invite/InviteLinkPage.tsx [hunk 2]
```diff
@@ -68,12 +77,12 @@ export default function InvitationPage ({ error, invite }: { error?: string, inv
             <PrimaryButton size='large' loading={!triedEager} onClick={openWalletSelectorModal}>
               Connect Wallet
             </PrimaryButton>
-            {/* <PrimaryButton size='large' loading={!triedEager} href={`/api/discord/oauth?redirect=${encodeURIComponent(window.location.href.split('?')[0])}&type=login`}>
+            <PrimaryButton size='large' loading={!triedEager} href={`/api/discord/oauth?redirect=${encodeURIComponent(window.location.href.split('?')[0])}&type=login`}>
```
Comment (mattcasey) [inline components/invite/InviteLinkPage.tsx:80]: remove the loading prop here

File: components/invite/InviteLinkPage.tsx [hunk 3]
```diff
@@ -49,17 +58,17 @@ export default function InvitationPage ({ error, invite }: { error?: string, inv
       </CenteredBox>
     );
   }
-  return (
+  return invite ? (
```
Comment (mattcasey) [inline components/invite/InviteLinkPage.tsx:61]: it's either error or invite that will exist. i just couldn't express it in the types, but i prefer not to add this logic check

File: components/invite/InviteLinkPage.tsx [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { InviteLinkPopulated } from 'lib/invites';
 import PrimaryButton from 'components/common/PrimaryButton';
 import { useUser } from 'hooks/useUser';
 import charmClient from 'charmClient';
+import { useSpaces } from 'hooks/useSpaces';
```
Comment (mattcasey) [inline components/invite/InviteLinkPage.tsx:13]: this is unused

File: components/invite/InviteLinkPage.tsx (hunk 1 above)
Comment (Devorein) [reply components/invite/InviteLinkPage.tsx:40]: If they have quite a lot of spaces I assume, but it's definitely a very rare case. Would cut off one extra API request to our backend and arguably a better UX IMO

File: components/invite/InviteLinkPage.tsx (hunk 3 above)
Comment (Devorein) [reply components/invite/InviteLinkPage.tsx:61]: Yep moved it with error handling to make it more readable

**Review by mattcasey — APPROVED** [review]
